lib/Extractor/ExprBuilder.cpp

```diff
@@ -376,6 +376,31 @@ Inst *ExprBuilder::getDemandedBitsCondition(Inst *I) {
                                    LIC->getInst(Inst::Eq, 1, {Res, Zero})});
     Result = LIC->getInst(Inst::And, 1, {Result, SignBits});
   }
+  if (!I->Range.isEmptySet() && !I->Range.isFullSet()) {
+    Inst *Lower = LIC->getConst(I->Range.getLower());
+    Inst *Upper = LIC->getConst(I->Range.getUpper());
+    Inst *NonWrappedRange = LIC->getConst(llvm::APInt(1, 0));
```

Comment:
you don't need these two temporaries, just add the appropriate condition directly to Result in the wrapped or unwrapped case